pkg/reconciler/autoscaling/hpa/resources/hpa.go

```diff
@@ -59,30 +59,46 @@ func MakeHPA(pa *autoscalingv1alpha1.PodAutoscaler, config *autoscalerconfig.Con
 		hpa.Spec.MinReplicas = &min
 	}
 
-	switch pa.Metric() {
-	case autoscaling.CPU:
-		if target, ok := pa.Target(); ok {
-			hpa.Spec.Metrics = []autoscalingv2beta1.MetricSpec{{
-				Type: autoscalingv2beta1.ResourceMetricSourceType,
-				Resource: &autoscalingv2beta1.ResourceMetricSource{
-					Name:                     corev1.ResourceCPU,
-					TargetAverageUtilization: ptr.Int32(int32(math.Ceil(target))),
+	if target, ok := pa.Target(); ok {
+		switch pa.Metric() {
+		case autoscaling.CPU:
+			hpa.Spec.Metrics = []autoscalingv2beta2.MetricSpec{{
+				Type: autoscalingv2beta2.ResourceMetricSourceType,
+				Resource: &autoscalingv2beta2.ResourceMetricSource{
+					Name: corev1.ResourceCPU,
+					Target: autoscalingv2beta2.MetricTarget{
+						Type:               "Utilization",
+						AverageUtilization: ptr.Int32(int32(math.Ceil(target))),
+					},
 				},
 			}}
-		}
 
-	case autoscaling.Memory:
-		if target, ok := pa.Target(); ok {
+		case autoscaling.Memory:
 			memory := resource.NewQuantity(int64(target)*1024*1024, resource.BinarySI)
-			hpa.Spec.Metrics = []autoscalingv2beta1.MetricSpec{{
-				Type: autoscalingv2beta1.ResourceMetricSourceType,
-				Resource: &autoscalingv2beta1.ResourceMetricSource{
-					Name:               corev1.ResourceMemory,
-					TargetAverageValue: memory,
+			hpa.Spec.Metrics = []autoscalingv2beta2.MetricSpec{{
+				Type: autoscalingv2beta2.ResourceMetricSourceType,
+				Resource: &autoscalingv2beta2.ResourceMetricSource{
+					Name: corev1.ResourceMemory,
+					Target: autoscalingv2beta2.MetricTarget{
+						Type:         "AverageValue",
```

Comment:
Is there a constant for `AverageValue` and `Utilization` maybe?

Reply:
:+1: